pkg/mover/rewrite_rules.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package mover
+
+import (
+	"errors"
+
+	"github.com/docker/distribution/reference"
```

Comment:
I had a peek at the `reference` package and I am wondering if this is not just an overkill to what we need, to check that the registry is a compliant URI and that the prefix is a path?

Reply:
I dig a little bit more and indeed it's tricky since regular url.ParseRequestURI is not compliant to the registry v2 format. For example OCI registry does not require a `scheme` and specifically removes special characters.

I guess that you were right and we will need to stick to your solution if we indeed want to frontload this validation.